nilearn/_utils/data_gen.py

```diff
@@ -866,93 +858,174 @@ def create_fake_bids_dataset(base_dir='',
         Creates a directory with dummy files.
 
     """
+    # FIXME entities are not ordered properly in the generated filenames
```

Comment:
Note that a proper way to create valid BIDS filename would be to rely on pybids path building capabilities.

This may be beyond the scope of this PR, but this would certainly simplify things.

Reply:
We decided in the past to _not_ rely on pybids ; this was partially the motivation for the development of ancp-bids. Given the development that's happened there in the last bit, it's likely worth revisiting ancp-bids to see if it can meet our needs soon. But we were pretty strongly against the heavier pybids dependency !

Reply:
okidoki, won't rehash this then.

FWIW there has been slow but steady progress on a pybids PR to reuse the ancp-bids machinery to index datasets to get a more than significant speed boost.
So if this was one in the argument against using bids, then I would suggest waiting before making a decision if we wanted to add a proper BIDS handling dependency into nilearn.

I am sure that in the meantime that the BIDS maintainer in me can survive with invalid BIDS filenames in dummy dataset. It will be hard but I will manage. #dramaqueen 👑 🎭